src/gtil/predict.cc

```diff
@@ -184,17 +199,23 @@ inline std::size_t PredictImpl(const treelite::ModelImpl<ThresholdType, LeafOutp
 namespace treelite {
 namespace gtil {
 
-std::size_t Predict(const Model* model, const DMatrix* input, float* output, bool pred_transform) {
+std::size_t Predict(const Model* model, const DMatrix* input, float* output, int nthread,
+                    bool pred_transform) {
+  // If nthread <= 0, then use all CPU cores in the system
+  if (nthread <= 0) {
```

Comment:
If you are using omp, it might be better to use utilities from omp instead of std thread.

Reply:
Fixed